homeassistant/components/light/mochad.py

```diff
@@ -86,11 +90,18 @@ def assumed_state(self):
         """X10 devices are normally 1-way so we have to assume the state."""
         return True
 
+    def _convert_brightness(self):
+        return int(
+            self._brightness * (float(self._brightness_levels) / 255.0))
+
     def turn_on(self, **kwargs):
         """Send the command to turn the light on."""
         self._brightness = kwargs.get(ATTR_BRIGHTNESS, 255)
+        mochad_brightness = self._convert_brightness()
         with mochad.REQ_LOCK:
-            self.device.send_cmd("xdim {}".format(self._brightness))
+            self.device.send_cmd("on")
+            self._controller.read_data()
+            self.device.send_cmd("dim {}".format(mochad_brightness))
```

Comment:
This dim command won't work properly.  If self._brightness is full on (255), the module is sent an 'on' followed by a 'dim 31', which basically turns it off.  The basic dim/bright X10 commands only have 5 bits for level.  Values higher than 31 will cause the field to wrap.  Devices that support more dim levels do them through extended commands.  Also, the basic commands are relative to the current state of the device, not an absolute value.  'dim 1' will reduce the devices level by one and 'bright 1' will increase it.

Reply:
ah, ok. I suspected that's how it worked. But, I don't have any hardware to test it on anymore, so I figured someone would tell me if I got it backwards, thanks. I'll update this patch to handle sending things as a change instead of a value.